homeassistant/components/isy994/services.py

```diff
@@ -0,0 +1,408 @@
+            if not hasattr(isy, "networking") or isy.networking is None:
+                continue
+            command = None
+            if address:
+                command = isy.networking.get_by_id(address)
+            if name:
+                command = isy.networking.get_by_name(name)
+            if command is not None:
+                await hass.async_add_executor_job(command.run)
+                return
+        _LOGGER.error(
+            "Could not run network resource command. Not found or enabled on the ISY."
+        )
+
+    async def async_send_program_command_service_handler(service):
+        """Handle a send program command service call."""
+        address = service.data.get(CONF_ADDRESS)
+        name = service.data.get(CONF_NAME)
+        command = service.data.get(CONF_COMMAND)
+        isy_name = service.data.get(CONF_ISY)
+
+        for config_entry_id in hass.data[DOMAIN]:
+            isy = hass.data[DOMAIN][config_entry_id][ISY994_ISY]
+            if isy_name and not isy_name == isy.configuration["name"]:
+                continue
+            program = None
+            if address:
+                program = isy.programs.get_by_id(address)
+            if name:
+                program = isy.programs.get_by_name(name)
+            if program is not None:
+                await hass.async_add_executor_job(getattr(program, command))
+                return
+        _LOGGER.error(
+            "Could not send program command. Not found or enabled on the ISY."
+        )
+
+    async def async_set_variable_service_handler(service):
+        """Handle a set variable service call."""
+        address = service.data.get(CONF_ADDRESS)
+        vtype = service.data.get(CONF_TYPE)
+        name = service.data.get(CONF_NAME)
+        value = service.data.get(CONF_VALUE)
+        init = service.data.get(CONF_INIT, False)
+        isy_name = service.data.get(CONF_ISY)
+
+        for config_entry_id in hass.data[DOMAIN]:
+            isy = hass.data[DOMAIN][config_entry_id][ISY994_ISY]
+            if isy_name and not isy_name == isy.configuration["name"]:
+                continue
+            variable = None
+            if name:
+                variable = isy.variables.get_by_name(name)
+            if address and vtype:
+                variable = isy.variables.vobjs[vtype].get(address)
+            if variable is not None:
+                await hass.async_add_executor_job(variable.set_value, value, init)
+                return
+        _LOGGER.error("Could not set variable value. Not found or enabled on the ISY.")
+
+    async def async_cleanup_registry_entries(service) -> None:
+        """Remove extra entities that are no longer part of the integration."""
+        entity_registry = await er.async_get_registry(hass)
+        config_ids = []
+        current_unique_ids = []
+
+        for config_entry_id in hass.data[DOMAIN]:
+            config_entry_owned_entities = er.async_entries_for_config_entry(
```

Comment:
We should name these `entries` and not `entities`. The latter is reserved for `Entity` instances.

Reply:
Renamed to `entries_for_this_config` to be explicit and clear.